app/history.py

```diff
@@ -19,44 +19,65 @@
 import app.log
 import cPickle as pickle
 import os
+import hashlib
 
-data = {}
-path = None
+fileHistory = {}
+userHistory = {}
+pathToHistory = None
+hasher = None
+checksum = None
+fileSize = 0
 
-def get(keyPath, default=None):
-  global data
-  cursor = data
-  for i in keyPath[:-1]:
-    cursor = cursor.setdefault(i, {})
-  return cursor.get(keyPath[-1], default)
+def get(keyPath, default={}):
+  return fileHistory.setdefault(keyPath[-1], default)
```

Comment:
This looks like it's only using the last element of keyPath. What's the advantage (or is this a whoops)?